File: Python/ValueChangeEvent.py
from caneasylib import *
import win32com.client as win32com

# Get CanEasyApplication instance
caneasy:ICanEasyApplication = getCanEasy()

class ChangeEvent:
    def OnValueChanged(self, _arg):
        arg = IChangeArg(_arg)
        valItem = IValue(arg.Item.DBItem)
        caneasy.MakeReport(f"{valItem.Name} = {valItem.Value}", constants.ReportTypeInformation)
 
@entry
class Application(BaseApplication):
    def OnStarting(self):        
        print("Example to handle value change events from database")

        # Create database item and assign reference
        sig = caneasy.CreateDatabaseItem()
        sig.StringRef = "//DB/Channel:Channel/Node:ECU/Msg:Message/Sig:Signal"
        
        # Activate change events
        recursive = False
        multithread = True
        sig.ActivateChangeEvents(recursive, multithread)
        win32com.WithEvents(sig, ChangeEvent)

File: CSharp/ValueChangeEvent.cs
using CanEasy;

namespace MyApplication
{
    public class Application : BaseApplication
    {
        protected override void OnStarting()
        {
            System.Console.WriteLine("Example to handle value change events from database");

            // Create database item and assign reference
            var sig = CanEasy.CreateDatabaseItem();
            sig.StringRef = "//DB/Channel:Channel/Node:ECU/Msg:Message/Sig:Signal";

            // Activate change events
            sig.ActivateChangeEvents(recursive:true, multithread:true);

            sig.OnValueChanged += (arg) =>
            {
                var item = (IValue)arg.Item.DBItem;
                CanEasy.MakeReport($"{item.Name} = {item.Value}", ReportType.ReportTypeInformation);
            };
        }
    }
}
